mqtt/mqtt-broker/src/error.rs

```diff
@@ -80,3 +83,19 @@ pub enum InitializeBrokerError {
     #[error("An error occurred  bootstrapping TLS")]
     Tls(#[source] native_tls::Error),
 }
+
+/// Authentication errors.
+#[derive(Debug, Error)]
+pub enum AuthenticationError {
+    #[error("failed to execute.")]
+    Execution,
+
+    #[error("failed to send request: {0}.")]
+    SendRequest(#[source] reqwest::Error),
```

Comment:
```suggestion
    SendRequest(#[from] reqwest::Error),
```
I'll give you From reqwest::Error to AuthenticationError for free